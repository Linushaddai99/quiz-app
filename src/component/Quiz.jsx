import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useEffect } from "react";
import getQuestions from "../redux/questionsApi";
import { useLocation } from "react-router-dom";
import Form from "./Form";

const Quiz = () => {
  const location = useLocation();
  const choice = location.state;

  const dispatch = useDispatch();
  const questions = useSelector((state) => state.questions.questions);
  const status = useSelector((state) => state.categories.status);
  console.log(questions);

  useEffect(() => {
    dispatch(getQuestions(choice));
  }, [dispatch, questions.length]);

  return (
    <div className="container question-card">
      <span>Category: </span>
      <span>{questions[0]?.category}</span>
      <Form questions={questions} status={status} />
    </div>
  );
};

export default Quiz;
