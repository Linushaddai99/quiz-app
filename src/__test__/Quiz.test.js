import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom';
import Quiz from '../component/Quiz'
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configStore.js';

describe('Quiz page', () => {
    it('Renders the quiz page correctly', () => {
        const { tree } = render(
            <Provider store={store}>
              <BrowserRouter>
                <Quiz />
              </BrowserRouter>
            </Provider>,
          );
          expect(tree).toMatchSnapshot();
    })
})