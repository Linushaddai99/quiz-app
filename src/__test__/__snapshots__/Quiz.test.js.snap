// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz page Renders the quiz page correctly 1`] = `
<div>
  <div
    class="container question-card"
  >
    <span>
      Category: 
    </span>
    <span />
    <div>
      <form>
        <div
          class="flex"
        >
          <h2>
            Here's Your Quiz
          </h2>
          <button
            type="button"
          >
            Restart
          </button>
        </div>
        <p>
          Oops!! there are no questions for your selected choices.
        </p>
        <button
          class="backbtn"
          type="button"
        >
          Generate new quesions
        </button>
      </form>
    </div>
  </div>
</div>
`;
